src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -0,0 +1,129 @@
+package de.zalando.aruha.nakadi.repository.kafka;
+
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.Properties;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import de.zalando.aruha.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+public class KafkaLocationManager {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaLocationManager.class);
+
+    private final String _BROKERS_IDS_PATH = "/brokers/ids";
+
+    @Autowired
+    private ZooKeeperHolder zkFactory;
+
+    private Brokers kafkaBrokerList;
+
+    private Properties kafkaProperties;
+
+    class Broker implements Comparable<Broker> {
+        final String host;
+        final Integer port;
+
+        public Broker(final String host, final Integer port) {
+            this.host = host;
+            this.port = port;
+        }
+
+        @Override
+        public int compareTo(Broker other) {
```

Comment:
My original idea was: if the list of brokers didn't changed in ZooKeeper we don't need to update it in the application.